utils/file-utils.js

```diff
@@ -0,0 +1,70 @@
+const fileType = require('file-type');
+const Stream = require('stream');
+const request = require('request-promise');
+const fs = require('fs');
+
+exports.file = (p) => {
+  if (exports.isUrl(p)) {
+    return request.get({ url: p, encoding: null });
+  }
+  return fs.createReadStream(p);
+};
+
+exports.isUrl = (s) => {
+  const regexp = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-/]))?/;
+  return regexp.test(s);
+};
+
+// Converts a stream to a buffer
+// Useful for locally calling a service
+// with a file
+exports.streamToBuffer = (s) => {
+  return new Promise((resolve) => {
+    const out = [];
+    s.on('data', (data) => {
+      out.push(data);
+    });
+    s.on('end', () => {
+      resolve(Buffer.concat(out));
+    });
+  });
+};
+
+// Converts buffers in response to file type
+exports.parseFileResponse = (response) => {
```

Comment:
Not sure what to do to simplify this. Maybe pull that parse reviver function outside of this function into a named one? Wondering if you came across this to do the buffer part? https://github.com/jprichardson/buffer-json Ours looks like it does more than that though....